dts/bindings/adc/adc-controller.yaml

```diff
@@ -12,3 +13,136 @@ properties:
+        required: true
+        description: Channel identifier.
+
+      zephyr,gain:
+        type: string
+        required: true
+        description: |
+          Gain selection:
+          - ADC_GAIN_1_6: x 1/6
+          - ADC_GAIN_1_5: x 1/5
+          - ADC_GAIN_1_4: x 1/4
+          - ADC_GAIN_1_3: x 1/3
+          - ADC_GAIN_2_5: x 2/5
+          - ADC_GAIN_1_2: x 1/2
+          - ADC_GAIN_2_3: x 2/3
+          - ADC_GAIN_4_5: x 4/5
+          - ADC_GAIN_1:   x 1
+          - ADC_GAIN_2:   x 2
+          - ADC_GAIN_3:   x 3
+          - ADC_GAIN_4:   x 4
+          - ADC_GAIN_6:   x 6
+          - ADC_GAIN_8:   x 8
+          - ADC_GAIN_12:  x 12
+          - ADC_GAIN_16:  x 16
+          - ADC_GAIN_24:  x 24
+          - ADC_GAIN_32:  x 32
+          - ADC_GAIN_64:  x 64
+          - ADC_GAIN_128: x 128
+        enum:
+          - "ADC_GAIN_1_6"
+          - "ADC_GAIN_1_5"
+          - "ADC_GAIN_1_4"
+          - "ADC_GAIN_1_3"
+          - "ADC_GAIN_2_5"
+          - "ADC_GAIN_1_2"
+          - "ADC_GAIN_2_3"
+          - "ADC_GAIN_4_5"
+          - "ADC_GAIN_1"
+          - "ADC_GAIN_2"
+          - "ADC_GAIN_3"
+          - "ADC_GAIN_4"
+          - "ADC_GAIN_6"
+          - "ADC_GAIN_8"
+          - "ADC_GAIN_12"
+          - "ADC_GAIN_16"
+          - "ADC_GAIN_24"
+          - "ADC_GAIN_32"
+          - "ADC_GAIN_64"
+          - "ADC_GAIN_128"
+
+      zephyr,reference:
+        type: string
+        required: true
+        description: |
+          Reference selection:
+          - ADC_REF_VDD_1:     VDD
+          - ADC_REF_VDD_1_2:   VDD/2
+          - ADC_REF_VDD_1_3:   VDD/3
+          - ADC_REF_VDD_1_4:   VDD/4
+          - ADC_REF_INTERNAL:  Internal
+          - ADC_REF_EXTERNAL0: External, input 0
+          - ADC_REF_EXTERNAL1: External, input 1
+        enum:
+          - "ADC_REF_VDD_1"
+          - "ADC_REF_VDD_1_2"
+          - "ADC_REF_VDD_1_3"
+          - "ADC_REF_VDD_1_4"
+          - "ADC_REF_INTERNAL"
+          - "ADC_REF_EXTERNAL0"
+          - "ADC_REF_EXTERNAL1"
+
+      zephyr,vref-mv:
+        type: int
+        required: false
+        description: |
+          This property can be used to specify the voltage (in millivolts)
+          of the reference selected for this channel, so that applications
+          can get that value if needed for some calculations.
+          For the internal reference, the voltage can be usually obtained with
+          a dedicated ADC API call, so there is no need to use this property
+          in that case, but for other references this property can be useful.
+
+      zephyr,acquisition-time:
+        type: int
+        required: true
+        description: |
+          Acquisition time.
+          Use the ADC_ACQ_TIME macro to compose the value for this property
+          or pass ADC_ACQ_TIME_DEFAULT to use the default setting for a given
+          hardware (e.g. when the hardware does not allow to configure the
+          acquisition time).
+
+      bipolar:
```

Comment:
Not really. The input does not necessarily need to be a pin. It can be some internal source, like prescaled VDD. The `adc_nrfx_adc` driver does not currently support this but the nRF ADC peripheral provides such possibility.

Reply:
Isn't this what the channel is for? Shouldn't input-positive/negative be reserved for differential channels?

Reply:
No, when `ADC_CONFIGURABLE_INPUTS` is selected, it is needed to specify `channel_id` and an input for it in `input_positive`, even for a single-ended channel. See [this description](https://github.com/zephyrproject-rtos/zephyr/blob/5af3c6ca905a4640ab96d7b6d5ddaa2536f23592/include/zephyr/drivers/adc.h#L111-L121).